implementation/endpoints/src/tcp_client_endpoint_impl.cpp

```diff
@@ -167,10 +167,15 @@ void tcp_client_endpoint_impl::connect() {
         // If specified, bind to device
         std::string its_device(configuration_->get_device());
         if (its_device != "") {
+#if defined(SO_BINDTODEVICE)
             if (setsockopt(socket_->native_handle(),
                     SOL_SOCKET, SO_BINDTODEVICE, its_device.c_str(), static_cast<socklen_t>(its_device.size())) == -1) {
                 VSOMEIP_WARNING << "TCP Client: Could not bind to device \"" << its_device << "\"";
             }
+#else
+            VSOMEIP_WARNING << "TCP Client: Could not bind to device \"" << its_device << "\""
```

Comment:
All the changes seem ok, but why not use the normal logging? VSOMEIP_WARNING << "tcp_client_endpoint::connect:.."  or  VSOMEIP_WARNING <<"tcp_client_endpoint::" << __func__ ""..

Reply:
I see your point and I had a few ideas on how to make this better as well, but I decided against it for the following reasons:

1. The warning log that was already present in the code uses the same formatting I have currently, I believe it makes sense to adhere to what's already there. I believe best practice is to keep logging style/formatting changes in separate commits in order to maintain a "prettier git log".
2. Having function signatures in logs is very useful for developers working on the library but it's too noisy for the user using the library. Your proposed log would make more sense for a DEBUG or TRACE level log. Perhaps in addition to the currently proposed WARNING log.

TL;DR I think your suggestion is good but only useful for the _developer_ and not-so-much the _user_. If you would like the addition of this  log as a DEBUG or TRACE, then please add it, but I believe our proposed WARNING should be there as it is.